### PR TITLE
Update doc to `–client-cert-auth`: it's recommended to enable this flag to avoid abuse of http2 for unauthenticated clients

### PR DESCRIPTION
**Pull request by ahrtr**
See discussion in https://github.com/etcd-io/etcd/issues/16740

cc @enj @tjungblu @chaochn47 @fuweid @serathius

### CODE REVIEW

**Comment by enj** [comment]
LGTM

**Review by jmhbnz — APPROVED** [review]
LGTM, thanks @ahrtr

File: content/en/docs/v3.4/op-guide/configuration.md [hunk 1]
```diff
@@ -275,7 +275,7 @@ The security flags help to [build a secure etcd cluster][security].
 + Enable client cert authentication.
 + default: false
 + env variable: ETCD_CLIENT_CERT_AUTH
-+ CN authentication is not supported by gRPC-gateway.
++ CN authentication is not supported by gRPC-gateway. Note if etcd is running as a public service, it's recommended to enable client cert authentication to avoid abuse of http2 for unauthenticated clients.
```
Comment (serathius) [inline content/en/docs/v3.4/op-guide/configuration.md:278]: Not sure about wording "public service". I think this is more about endpoint availability within a network. Even if the network is trusted, there is a risk that attacker will penetrate it.

```suggestion
+ CN authentication is not supported by gRPC-gateway. Note that running etcd without client cert authentication expose its endpoint to http2 protocol abuse from unauthenticated clients (CVE-2023-44487).
```

Comment (ahrtr) [reply content/en/docs/v3.4/op-guide/configuration.md]: Running etcd as a public service definitely is more vulnerable the the CVE. 

Updated anyway & thx.

**Review by fuweid — APPROVED** [review]
LGTM

**Comment by fuweid** [comment]
Maybe we can add note to `peer-client-cert-auth` as well.

**Comment by ahrtr** [comment]
> Maybe we can add note to `peer-client-cert-auth` as well.

Yes, it's a good point. Updated & thx.

**Comment by jberkus** [comment]
I'd take out the "Note" on all of these; "It's recommended" reads much better.  But really, I'm waiting for someone else to look at the latest version before merging.

**Comment by ahrtr** [comment]
> I'd take out the "Note" on all of these; "It's recommended" reads much better.

thx & updated.

**Review by serathius — APPROVED** [review]

**Comment by ahrtr** [comment]
Updated the description to make it more generic.  Note `CVE-2023-44487` is just one type of attacks.

**Pull request merged**